### DISCUSSION TITLE
The project is looking for new maintainers

### DISCUSSION THREAD

**Discussion by jbe-dw** [ngoduykhanh/PowerDNS-Admin#1283] [Announcements]
Dear community,

I had recent changes in my life that drew me away from this project. The other maintainers have changed their priorities. As consequence, the project is left unmaintained for weeks. I tried to recruit more person in the project without any success, and our small team could fix and improve a little bit PDA, but can no longer succeed at this task.

If any adopter wants to step up to continue, I'd be happy to let him/her in. I hope so, because I'd be sad if this project would die.

Please let me know if you want to take up the torch.

Regards
J.

**Comment by AzorianMatt** [comment]
If all efforts continue to fail, just remember I'm still around and my original proposal still stands to take ownership of the project.

I could at least keep efforts open for new maintainers but could not keep a promise that I would personally keep the current version alive without a change in contributions.

My concern is that the reality I painted in a previous discussion is coming true and that no one is interested in contributing at its current level. After further thought on the project over the last year, I decided my direction would not be to implement a highly complicated view layer on the client side in efforts to keep the solution simple.

My current thoughts for the project would be to do a lot of cleanup on the current organization methods so that the project would have uniformed implementation across the board as it relates to code organization and separation.

I have a fairly old version of the project deployed in multiple production scenarios so it is useful to us but I would love to see some aggressive improvement. The only way I see myself personally getting there with my current capacity and motivations, is a bit of an overhaul.

**Reply by jbe-dw** [reply]
Hello,

I agree with you on this point: the code base comes from a personal project that was shared and got larger and larger. There has been no rules in the repository and contributions as the git log has become horribly messy. It's a pain to get in as things have been implemented differently here and there.

I asked the author to be part of the project when he went away. I told him that I would help for management but not to fix code or add functionalities. In the end, the applications were not many and efforts were not sustained over the weeks. We seek to fix bugs to an honorable v1 before thinking about a fresh v2, but we never arrived there.

 The thing is that rebooting the project is a huge load of work that is hard to manage on your own, so it should be considered with reason. I told you that I loved to see a core application with a strong API that could be consumed by a modern GUI AJAX calls. I'd love to have a multiple backend management too, which would require a rewrite too.

Personally, I use it in my enterprise, mainly for the API Proxy delegating permissions. I have built a Terraform provider so that management is made by code. However, giving users access to a GUI is quite comforting for them.

I am convinced that PDA is a great tool that can strongly enforce PowerDNS as a cloud native DNS in enterprises, so enterprise grade features like authentication methods should not be forgotten. And I'd like to be sure that it won't. Can you elaborate your thought ? When you say:

>  I decided my direction would not be to implement a highly complicated view layer on the client side in efforts to keep the solution simple.

**Reply by AzorianMatt** [reply]
Hi again @jbe-dw,

You continue to make valid points, especially as it relates to the enterprise level authentication features. I understand that those features are likely of important use to many users (admittedly, I'm not one of them currently). I recognize the need to maintain that support, especially if a large overhaul were to occur.

To elaborate on that particular thought, here is where my head is at. I am thinking as part of an overhaul, a more traditional vanilla JS approach would be used much like we did before so many of the fancy view frameworks really took hold. I'm referring to the good ole' days of simple jQuery with plugins as needed. Basic jQuery would be used for implementation of minimal, required JS features and plugins such as jQuery Validate would be used for user input validation on forms. This would support establishing some form of standard without implementing anything that has any real learning curve to it. If nothing else, it could serve as a stepping stone to the implementation of a view framework down the road but still solve some of the current issues today without a lot of work.

**Reply by jbe-dw** [reply]
Ok I get it. As I already told you, I have no experience in modern JS (coded some stuff when JQuery was not born yet) so I'm not able to give a technical opinion about that.

I summoned every candidate in a chatroom, I won't give all of you the access and leave, but I won't be much part of decisions that the new team would take.

See you there !

**Reply by OrdenKrieger** [reply]
I would love to do a full rewrite of the frontend with [Vue.js](https://vuejs.org/) and [Tailwind CSS](https://tailwindcss.com/). But I don't think I'll ever have time to do so. Anyways, if anyone would start an overhaul of the frontend, I would be happy to help in my free time. At work as @AdvanticGmbH I got only resources to help with major issues. ;)

**Reply by AzorianMatt** [reply]
@OrdenKrieger we would love to have more help in the near future as efforts are ramping up to get the project alive again. If you get to where you can commit to contributions, then feel free to email me at admin@powerdnsadmin.org and we can get you setup with access to the project's Mattermost deployment. Thanks @OrdenKrieger!

**Comment by jasonmacer** [comment]
I thought I had said something on this thread., but it is highly likely that I forgot to hit "save," now that I'm thinking about it. I have a bad habit of doing that on my phone, and I initially saw this thread there when I was browsing redit.

I, like @AzorianSolutions, have several older version deployed in production, and I would love to also see some improvements. 

I'm here to help as well.

**Comment by hydromike** [comment]
I am loving this project, I don't have any front end javascript experience.  I am more a python/django/backend developer.  Let me know what I can do. *EDIT* I was going thru the code last night and I am thinking that their could be more documentation on using the admin interface.  Other than that I do love htmx it does help reduce the amount of javascript that has to be used for a responsive UI.

**Comment by AzorianMatt** [comment]
Hi everyone,

For those of you still interested in contributing to the rebirth of this project and many great things to come, then please join us in the project's new Mattermost deployment (OSS Slack essentially). It is by invite only so to get an invite, please email me at "admin@powerdnsadmin.org" and I will provide a registration link for you.
